lib/sedna/service/server/base.py

```diff
@@ -12,16 +12,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import contextlib
+import time
+import threading
+import asyncio
+
 import uvicorn
 from fastapi.middleware.cors import CORSMiddleware
 
 from sedna.common.log import LOGGER
 from sedna.common.utils import get_host_ip
 
 
+class Server(uvicorn.Server):
+    def install_signal_handlers(self):
+        pass
+
+    @contextlib.contextmanager
+    def run_in_thread(self):
+        thread = threading.Thread(target=self.run)
+        thread.start()
+        try:
+            while not self.started:
+                time.sleep(1e-3)
```

Comment:
So here wait server to be started, `while and time.sleep` method is not elegant, can find a better one?

Reply:
A question: does it need to wait server to be started?

Reply:
In this PR, multithreaded  are used to monitor the service status. Since `uvicorn` (backend of http/ws server) is inherently multithreaded, there's the question of A/ waiting for the server to startup , and B/ triggering the server shutdown and have it clean up gracefully.

Reply:
look like no need to do A(waiting for the server to startup) here: because B(triggering the server shutdown) depends on A, A is guaranteed to happend when you want to do B

Reply:
fix